core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestApplyTableScanRedirection.java

```diff
@@ -105,6 +105,30 @@ public void testDoesNotFire()
         }
     }
 
+    @Test
+    public void testDoesNotFireForDeleteTableScan()
```

Comment:
Would it be possible to have an integration test as well?
there can be other problem(s) in delete flow; it's kind of special

Reply:
> Would it be possible to have an integration test as well?

We don't have a connector implementation yet that supports redirections and deletes